app/helpers/course/assessment/answer/programming_helper.rb

```diff
@@ -37,7 +37,7 @@ def insert_annotations(code_block_html, programming_answer_file)
     document = Nokogiri::XML.parse(code_block_html)
 
     programming_answer_file.annotations.each do |annotation|
-      insert_annotation_block(document, annotation)
+      insert_annotation_block(document, annotation) unless annotation.posts.empty?
```

Comment:
I put it here because if we just leave it as `insert_annotation_block(document, annotation)`, whoever is reading the code will expect a block to be inserted and it will come as a surprise if no block is inserted.

I would agree with your approach if we were doing something like `render programming_answer_file.annotations` here instead.